app/services/oai_query_string_service.rb

```diff
@@ -16,10 +16,12 @@ def self.process_from_time(full_index)
     from_time
   end
 
-  def self.process_string(saved_resumption_token, oai_set, from_time, to_time)
+  def self.process_string(saved_resumption_token, oai_set, institution, from_time, to_time, single_record)
     # resume from last harvested
     return "?verb=ListRecords&resumptionToken=#{saved_resumption_token}" if saved_resumption_token.present?
-    # start fresh harvest
+    # start a single record harvest
+    return "?verb=GetRecord&identifier=oai:alma.#{institution}:#{oai_set}&metadataPrefix=marc21&until=#{to_time}#{from_time}" if single_record
```

Comment:
Does the institution ever change? If not, does `ENV` not seem to work in this file?